plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClientModule.java

```diff
@@ -53,33 +54,75 @@ public void configure(Binder binder)
         configBinder(binder).bindConfig(OracleConfig.class);
         newOptionalBinder(binder, Key.get(int.class, MaxDomainCompactionThreshold.class)).setBinding().toInstance(ORACLE_MAX_LIST_EXPRESSIONS);
         newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);
+        binder.bind(RetryingConnectionFactory.class).in(Scopes.SINGLETON);
     }
 
     @Provides
     @Singleton
     @ForBaseJdbc
-    public static ConnectionFactory connectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider, OracleConfig oracleConfig, OpenTelemetry openTelemetry)
-            throws SQLException
+    public ConnectionFactory connectionFactory(
+            OracleConfig oracleConfig,
+            OraclePoolConnectionFactory oraclePoolConnectionFactory,
+            RetryingConnectionFactory retryingConnectionFactory)
     {
-        Properties connectionProperties = new Properties();
-        connectionProperties.setProperty(OracleConnection.CONNECTION_PROPERTY_INCLUDE_SYNONYMS, String.valueOf(oracleConfig.isSynonymsEnabled()));
-        connectionProperties.setProperty(OracleConnection.CONNECTION_PROPERTY_REPORT_REMARKS, String.valueOf(oracleConfig.isRemarksReportingEnabled()));
-
         if (oracleConfig.isConnectionPoolEnabled()) {
```

Comment:
You can use `ConditionalModule`:

```
     install(conditionalModule(
                OracleConfig.class,
                OracleConfig:: isConnectionPoolEnabled,
                innerBinder -> innderBinder.bind(ConnectionFactory.class, ForBaseJdbc.class).to(RetryingConnectionFactory.class).in(SINGLETON),
                innerBinder -> innderBinder.bind(OraclePoolConnectionFactory.class, ForBaseJdbc.class).to(RetryingConnectionFactory.class).in(SINGLETON)));
```

Reply:
I could. I was thinking whether I could do it in a neater way.

However, the one you proposed seems overengineering (many more lines of code, much harder to read).

I can replace with you proposal - have no attachment to one solution or the other. But I also like to give you a chance of re-thinking the proposal. Let me know your decision.

Reply:
`ConditionalModule` I used quite frequently.

Reply:
I tried to do that but I am getting errors that I don't understand. I left commented out code in case you wanted to help.